### PR TITLE
[Model] Creates Quote Model

### PR DESCRIPTION
**Pull request by vinibr3**
Creates quote model and its tests. Fix bug in number address's Factory.

### CODE REVIEW

**Review by RodrigoNM — DISMISSED** [review]

File: spec/models/person_spec.rb [hunk 1]
```diff
@@ -3,22 +3,23 @@
...
-  it {  is_expected.to validate_presence_of(:email) }
-  it {  is_expected.to_not allow_value("jonh@example").for(:email) }
-  it {  is_expected.to_not allow_value("@example.com").for(:email) }
-  it {  is_expected.to validate_uniqueness_of(:cpf) }
-  it {  is_expected.to_not allow_value(Faker::IDNumber.invalid).for(:cpf) }
-  it {  is_expected.to validate_uniqueness_of(:cnpj) }
-  it {  is_expected.to_not allow_value(Faker::IDNumber.invalid).for(:cpf) }
-  it {  is_expected.to validate_uniqueness_of(:rg) }
-  it {  is_expected.to validate_length_of(:uf_expediter_rg) }
+  it { is_expected.to have_many(:telephones) }
```
Comment (kaiomagalhaes) [inline spec/models/person_spec.rb:6]: please split the validations, like:

have_many
have_many
(empty line)
belong_to
belong_to
(empty_line)
....

File: spec/models/quote_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+require "rails_helper"
+
+RSpec.describe Quote, type: :model do
+  let(:quote) { create(:quote) }
+
+  it { is_expected.to validate_length_of(:code).is_at_most(255) }
```
Comment (kaiomagalhaes) [inline spec/models/quote_spec.rb:6]: please split the validations, like:

have_many
have_many
(empty line)
belong_to
belong_to
(empty_line)
.

File: spec/models/quote_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  it { is_expected.to belong_to(:quote_type) }
+  it { is_expected.to belong_to(:person) }
+  it { is_expected.to have_db_index(:user_id) }
+  it { is_expected.to have_db_index(:quote_status_id) }
+  it { is_expected.to have_db_index(:quote_type_id) }
+  it { is_expected.to have_db_index(:person_id) }
+  it { is_expected.to have_one(:order) }
+
+  context "when validate calculate total value" do
+    let(:discount) { quote.discount.to_d }
```
Comment (kaiomagalhaes) [inline spec/models/quote_spec.rb:30]: are these values specific for this context? if not please put this let at the top of the class

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: spec/models/quote_spec.rb (hunk 3 above)
Comment (vinibr3) [reply spec/models/quote_spec.rb:30]: Yes. These are.

**Review by kaiomagalhaes — APPROVED** [review]

**Review by matcheus — APPROVED** [review]

**Pull request merged**